framework/src/meshgenerators/SideSetsFromBoundingBoxGenerator.C

```diff
@@ -80,6 +103,22 @@ SideSetsFromBoundingBoxGenerator::generate()
   bool found_side_sets = false;
   const bool inside = (_location == "INSIDE");
 
+  // Get the list of boundary ids from the boundary names
+  auto boundary_ids = MooseMeshUtils::getBoundaryIDs(*mesh, _boundaries_old, false);
+
+  // Check that the old boundary ids/names exist in the mesh
+  for (std::size_t i = 0; i < boundary_ids.size(); ++i)
+    if (boundary_ids[i] == Moose::INVALID_BOUNDARY_ID)
+      paramError(
+          "boundaries", "The boundary '", _boundaries_old[i], "' was not found within the mesh");
+
+  // Attempt to get the new boundary id from the name
+  auto boundary_id_new = MooseMeshUtils::getBoundaryID(_boundary_new, *mesh);
+
+  // If the new boundary id/name is not valid, make it instead
+  if (boundary_id_new == Moose::INVALID_BOUNDARY_ID)
+    boundary_id_new = cast_int<boundary_id_type>(*(boundary_info.get_boundary_ids().rbegin()) + 1);
```

Comment:
this is the right code, but we would want a utility for this because it s tricky
could you please add to mooseMeshUtils the equivalent of this:
```
/**
 * Checks input mesh and returns max(block ID) + 1, which represents
 * a block ID in the mesh that is not currently in use
 * @param input mesh over which to compute the next free block id
 */
SubdomainID getNextFreeSubdomainID(MeshBase & input_mesh);

```
for BoundaryIDs